### ISSUE TITLE
custom `expectedResponse` not working when using Combine publisher mode

### ISSUE THREAD

**Issue by zenghaojim33** [rwbutler/Connectivity#74]
**Have read the known issues? **
[ ] Yes, I have read and am familiar with the list of [known issues](https://github.com/rwbutler/Connectivity#known-issues).

**Have tested on a physical device? **
[ ] Yes, I have tested on physical iOS device and the reported issue is still present.

**Does code retain the `Connectivity` object? **
[ ] Yes, my implementation has a strong reference to the `Connectivity` object so that it is not deallocated during use.

**Describe the bug**
custom expectedResponse value is not working

**To Reproduce**
1. Go to `CombineViewController` , configure the publisher as below:
```
let publisher = Connectivity.Publisher(
            configuration:
                    .init(
                        responseValidator: ConnectivityResponseContainsStringValidator(expectedResponse: "message"),
                        validatioMode: .containsExpectedResponseString)
                .configureURLSession(.default)
        ).eraseToAnyPublisher()
```
3. run the notifier

**Expected behavior**

The connection should be considered lost because the expected reponse has been configured as `message` instead of `Success`. However the `ConnectivityResponseValidator` is still taking the default `Success` value.

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by zenghaojim33** [comment]
Hi @rwbutler I tried with the latest version ( and I found the bug in the previous version during the day ).  Turns out that the validator factory is not taking the `expectedResponseString` from the one passed into `responseValidator`, but the one declared in `Connectivity` class. Later in `manufacture()` it returns a new validator with that default value. Didn't look into it but I assume that same behavior will happen to `expectedResponseRegEx` as well. Now I am switch to original usage and explictly assign the value which works for me.

<img width="397" alt="image" src="https://user-images.githubusercontent.com/8848084/224059528-5d56f891-ce19-4f78-9477-60efef0445a1.png">
